### PR TITLE
Fixed some grammers and typos in comments & readMe files

### PR DESCRIPTION
**Pull request by Saran-pariyar**
I checked the code and some readMe files, code comments, etc had some typos and grammar mistakes so I fixed them

### Pull Request checklist ###

- [ ] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [x] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work
  - Documentation should be added to [The Glean Book](https://mozilla.github.io/glean/book/index.html) when making changes to Glean's user facing API
    - When changes to the Glean Book are necessary, link to the corresponding PR on the [`mozilla/glean`](https://github.com/mozilla/glean) repository
  - Documentation should be added to [the Glean.js developer documentation](https://github.com/mozilla/glean.js/tree/main/docs) when making internal code changes

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -97,7 +96,7 @@ The `entry/` folder contains the main entry points for the Glean.js package per
 For example, when a user does `import Glean from @mozilla/glean/webext` it's the `entry/webext.ts`
 file that they are getting and not `core/glean.ts`.
 
-The main difference between each platform's file is that a different `Platform` implementation is
+The main difference between each platform's files is that a different `Platform` implementation is
```
Comment (rosahbruno) [inline ARCHITECTURE.md:99]: Let's leave this how it was. If you look inside of the `glean/src/entry` folder, each platform has its own file, not files.

```suggestion
The main difference between each platform's file is that a different `Platform` implementation is
```

**Review by rosahbruno — CHANGES_REQUESTED** [review]
Hello and welcome - thank you for your contribution!

Most of these changes look great, just a few small, nitpicky things and then we can get this merged!

**Comment by Saran-pariyar** [comment]
Corrected mistake  and made some spelling changes

**Review by rosahbruno — APPROVED** [review]
Nice job! Thanks a lot and congrats on your first contribution!!

**Pull request merged**